Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -180,6 +185,22 @@ void FunctionDefinitionWriter::visit(AST::Structure& structDecl)
 }
 
 void FunctionDefinitionWriter::visit(AST::Variable& variable)
+{
+    serializeVariable(variable);
+    m_stringBuilder.append(";\n");
+}
+
+void FunctionDefinitionWriter::visitGlobal(AST::Variable& variable)
+{
+    if (variable.flavor() != AST::VariableFlavor::Override)
+        return;
+
+    m_stringBuilder.append("constant ");
+    serializeVariable(variable);
```

Comment:
Where is the validation that only certain types are allowed to be function constants?

Where is this tested?

Reply:
We have basically no validation of that sort yet. We could test by matching the metal output, but behavior tests won't work yet since we still need one more patch to get overrides to actually work.